финальная домашка по разделу PetShop

EkaterinaMaximovaQA opened the pull request:
- в файл conftest - доавлена новая фикстура для размещения заказа,в Swagger раздел Store

 - создан новый Python file /test_store.py\

- создан класс Store для автоматизации тестов:
+ Проверка размещения заказа в магазине
+ Получение информации о заказе по ID
+ Удаление заказа по ID
+ Попытка получить информацию о несуществующем заказе
+ Получение инвентаря магазина

ссылка на Allure отчет http://5.181.109.28:8080/job/KateMaxPetShopApiTests/allure/#suites

EkaterinaMaximovaQA committed:
- в файл conftest - доавлена новая фикстура для размещения заказа,в Swagger раздел Store
 - создан новый Python file /test_store.py\

- создан класс Store для автоматизации тестов:
+ Проверка размещения заказа в магазине
+ Получение информации о заказе по ID
+ Удаление заказа по ID
+ Попытка получить информацию о несуществующем заказе
+ Получение инвентаря магазина

dmitryvshivtsev commented on test/test_store.py:
```diff
@@ -0,0 +1,77 @@
+                response = requests.post(url=f"{BASE_URL}/store/order",json=create_order)
+                response_json = response.json()
+            with allure.step("Проверка статуса ответа и данных заказа"):
+                assert response.status_code == 200
```
тут тоже можно написать схему, как в pet

EkaterinaMaximovaQA replied:
done

dmitryvshivtsev commented on test/test_store.py:
```diff
@@ -0,0 +1,77 @@
+        "expected_data, expected_status_code",
+        [
+            ({"approved": 57, "delivered": 50}, 200) #в Swagger в примере эндпоинта данных больше чем в требованиях самого теста, если это так и должно быть - ОК,если инфа устарела - я добавлю...ну либо я что - то не так поняла как делать +)))
+        ]
```
Параметризация здесь не нужна. Просто проверь формат ответа. В идеале с помощью схемы, а не просто dict

EkaterinaMaximovaQA replied:
done

EkaterinaMaximovaQA committed:
- внесены правки по PR от Димы
- создан файл с json-schema для размещения заказа
- создан файл с json-shema для получения инвентаря

EkaterinaMaximovaQA commented:
перезапуск тестов:
http://5.181.109.28:8080/job/KateMaxPetShopApiTests/allure/#suites
